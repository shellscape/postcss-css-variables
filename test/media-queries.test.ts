import { fixture, process } from './helpers';

describe('media queries', () => {
  test('should add rule declaration of property in @media', async () => {
    const css = await fixture('media/query');
    const result = await process(css, { preserveAtRulesOrder: true });
    expect(result).toMatchSnapshot();
  });

  test('should work with @media, preserving rule order', async () => {
    const css = await fixture('media/preserve-rule-order');
    const result = await process(css, { preserveAtRulesOrder: true });
    expect(result).toMatchSnapshot();
  });

  test('should work with nested @media', async () => {
    const css = await fixture('media/nested');
    const result = await process(css, { preserveAtRulesOrder: false });
    expect(result).toMatchSnapshot();
  });

  test('should work with nested @media, preserving rule order', async () => {
    const css = await fixture('media/nested-preserver-rule-order');
    const result = await process(css, { preserveAtRulesOrder: true });
    expect(result).toMatchSnapshot();
  });

  test('preserves variables in @media when `preserve` is `true`', async () => {
    const css = await fixture('media/preserve-variables');
    const result = await process(css, { preserveAtRulesOrder: true });
    expect(result).toMatchSnapshot();
  });

  test('should not double media queryies', async () => {
    const css = await fixture('media/double');
    const result = await process(css, { preserveAtRulesOrder: true });
    expect(result).toMatchSnapshot();
  });
});
