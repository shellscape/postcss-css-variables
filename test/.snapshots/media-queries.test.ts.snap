// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`media queries > preserves variables in @media when \`preserve\` is \`true\` 1`] = `".box{width:100px}@media (max-width: 1000px){.box{width:200px}}"`;

exports[`media queries > should add rule declaration of property in @media 1`] = `".box{width:100px}@media (max-width: 1000px){.box{width:200px}}"`;

exports[`media queries > should not double media queryies 1`] = `"@media (prefers-color-scheme: dark){.dark\\:bg-black{background-color:rgb(0 0 0/1)}@media (prefers-color-scheme: dark){.dark\\:bg-black{background-color:rgb(0 0 0/1)}}}"`;

exports[`media queries > should work with @media, preserving rule order 1`] = `".box{width:150px}@media (min-width: 1000px){.box{width:300px}}@media (min-width: 1250px){.box{width:450px}}"`;

exports[`media queries > should work with nested @media 1`] = `".box{width:150px}@media print{@media (max-width: 1250px){@media (max-width: 1000px){.box{width:450px}}}}@media print{@media (max-width: 1250px){.box{width:300px}}}"`;

exports[`media queries > should work with nested @media, preserving rule order 1`] = `".box{width:150px}@media print{@media (min-width: 1000px){.box{width:300px}}}@media print{@media (min-width: 1000px){@media (min-width: 1250px){.box{width:450px}}}}"`;
